lms/djangoapps/instructor/views/api.py

```diff
@@ -2625,24 +2628,77 @@ def enable_certificate_generation(request, course_id=None):
     return redirect(_instructor_dash_url(course_key, section='certificates'))
 
 
-#---- Gradebook (shown to small courses only) ----
-@cache_control(no_cache=True, no_store=True, must_revalidate=True)
-@require_level('staff')
-def spoc_gradebook(request, course_id):
+def calculate_page_info(offset):
     """
-    Show the gradebook for this course:
-    - Only shown for courses with enrollment < settings.FEATURES.get("MAX_ENROLLMENT_INSTR_BUTTONS")
-    - Only displayed to course staff
+    This method calculates query offsets for next and previous page, current page number on give
+    current query offset.
+
+    :param offset: offset for database query
+    :return: tuple consist of page number, query offset for next and previous pages and valid offset
     """
-    course_key = SlashSeparatedCourseKey.from_deprecated_string(course_id)
-    course = get_course_with_access(request.user, 'staff', course_key, depth=None)
 
-    enrolled_students = User.objects.filter(
+    # validate offset
+    if not isinstance(offset, int) and not offset.isdigit():
+        offset = 0
+    elif offset < 0:
+        offset = 0
+    else:
+        offset = int(offset)
+
+    next_offset = offset + MAX_STUDENTS_PER_PAGE_GRADE_BOOK
+    if next_offset < 0:
+        next_offset = None
+
+    previous_offset = offset - MAX_STUDENTS_PER_PAGE_GRADE_BOOK
+    if previous_offset < 0:
+        previous_offset = None
+
+    page_num = ((offset / MAX_STUDENTS_PER_PAGE_GRADE_BOOK) + 1)
+    return previous_offset, next_offset, page_num, offset
+
+
+def get_grade_book_page(request, course, course_key):
+    """
+    Get student records per page along with page information i.e current page, total pages and
+    offset information.
+    """
+    offset = request.GET.get('offset', 0)
+    previous_offset, next_offset, page_num, offset = calculate_page_info(offset)
+
+    # calculate total number of pages
+    total_students = User.objects.filter(
         courseenrollment__course_id=course_key,
         courseenrollment__is_active=1
-    ).order_by('username').select_related("profile")
-
-    # possible extension: implement pagination to show to large courses
+    ).order_by('username').select_related("profile").count()
```

Comment:
even if there is not much computational overhead on repeating 
```
User.objects.filter(
    courseenrollment__course_id=course_key,
    courseenrollment__is_active=1
).order_by('username').select_related("profile")
```

I think you can assign that to a variable (let's say `students_queryset`) at the beginning and then apply the `.count()` and the slicing.